### PR TITLE
Add support for user, workdir, and env docker parameters

### PR DESCRIPTION
**Pull request by pschorf**
## Changes proposed in this PR
- Add support for `workdir`, `env`, and `docker` parameters

## Why are we making these changes?
Increase support for the existing mesos API

### CODE REVIEW

File: integration/tests/cook/test_basic.py [hunk 1]
```diff
@@ -2639,6 +2698,32 @@ def test_kubernetes_disallowed_volumes(self):
...
+            self.skipTest(f'Requires {bad_var_name} to be in the disallowed-var-names config')
+        docker_image = util.docker_image()
+
+        container = {'type': 'docker',
+                     'docker': {'image': docker_image,
+                                'network': 'HOST',
+                                'force-pull-image': False,
+                                'parameters': [{'key': 'env',
+                                                'value': f'{bad_var_name}=set'}]}}
+        command = 'bash -c \'if [ -z ${BADVAR+x} ]; then exit 0; else exit 1; fi\''
```
Comment (dposada) [inline integration/tests/cook/test_basic.py]: What does `BADVAR+x` do?

File: scheduler/src/cook/kubernetes/api.clj [hunk 2]
```diff
@@ -322,22 +393,29 @@
...
+                           (into env)
+                           (into (param-env-vars parameters))
+                           (into workdir-env-vars)
+                           filter-env-vars))
     (.setImage container image)
 
     (.putRequestsItem resources "memory" (double->quantity (* memory-multiplier mem)))
     (.putLimitsItem resources "memory" (double->quantity (* memory-multiplier mem)))
     (.putRequestsItem resources "cpu" (double->quantity cpus))
+    (.putLimitsItem resources "cpu" (double->quantity cpus))
```
Comment (dposada) [inline scheduler/src/cook/kubernetes/api.clj]: Should we add a comment about why we're doing this?

File: integration/tests/cook/test_basic.py (hunk 1 above)
Comment (pschorf) [reply integration/tests/cook/test_basic.py]: That is equivalent to x if `BADVAR` is set to anything (including empty string) and still null if it is not set. It seems like the best way to test for a variable not being set.

**Comment by dposada** [comment]
Will merge when green

**Pull request merged**